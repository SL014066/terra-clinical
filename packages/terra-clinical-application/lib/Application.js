'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _terraBase = require('terra-base');

var _terraBase2 = _interopRequireDefault(_terraBase);

var _terraClinicalAppDelegate = require('terra-clinical-app-delegate');

var _terraClinicalAppDelegate2 = _interopRequireDefault(_terraClinicalAppDelegate);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  app: _terraClinicalAppDelegate2.default.propType,
  children: _react.PropTypes.node
};

var Application = function Application(_ref) {
  var app = _ref.app,
      children = _ref.children,
      customProps = _objectWithoutProperties(_ref, ['app', 'children']);

  return _react2.default.createElement(
    _terraBase2.default,
    _extends({}, customProps, { className: (0, _classnames2.default)([customProps.className, 'terraClinical-Application']) }),
    _react2.default.Children.map(children, function (child) {
      return _react2.default.cloneElement(child, { app: app });
    })
  );
};

Application.propTypes = propTypes;

exports.default = Application;