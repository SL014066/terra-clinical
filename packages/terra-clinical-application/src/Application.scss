@import '~terra-mixins';

.terraClinical-Application {
  height: 100%;
}
