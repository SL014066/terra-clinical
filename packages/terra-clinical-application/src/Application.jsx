import React, { PropTypes } from 'react';
import classNames from 'classnames';
import Base from 'terra-base';
import AppDelegate from 'terra-clinical-app-delegate';

import './Application.scss';

const propTypes = {
  app: AppDelegate.propType,
  children: PropTypes.node.isRequired,
};

const Application = ({ app, children, ...customProps }) => {
  let childrenToRender = children;

  if (app) {
    childrenToRender = React.Children.map(children, child => (
      React.cloneElement(child, { app })
    ));
  }

  return (
    <Base {...customProps} className={classNames([customProps.className, 'terraClinical-Application'])}>
      {childrenToRender}
    </Base>
  );
};

Application.propTypes = propTypes;

export default Application;
