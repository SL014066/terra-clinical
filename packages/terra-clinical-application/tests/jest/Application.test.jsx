import React from 'react';
import Application from '../../src/Application';

describe('Application', () => {
  const defaultRender = <Application />;

  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  // Prop Tests
  it('should use the default value when no value is given', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.find('.terraClinical-Application').text()).toEqual('defualt');
  });

  // Structure Tests
  it('should have the class terraClinical-Application', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.prop('className')).toContain('terraClinical-Application');
  });
});
