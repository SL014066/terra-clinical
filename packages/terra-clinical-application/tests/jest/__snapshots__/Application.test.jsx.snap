exports[`test should render children with app prop 1`] = `
<Base
  className="terraClinical-Application">
  <div
    app={
      AppDelegateInstance {
        "closeDisclosure": undefined,
        "disclose": undefined,
        "dismiss": undefined,
        "goBack": undefined,
        "maximize": undefined,
        "minimize": undefined,
      }
    }>
    TestApp
  </div>
</Base>
`;

exports[`test should render custom properties on component 1`] = `
<Base
  className="custom-class terraClinical-Application"
  id="my-id">
  <div>
    TestApp
  </div>
</Base>
`;

exports[`test should render default component 1`] = `
<Base
  className="terraClinical-Application">
  <div>
    TestApp
  </div>
</Base>
`;
